treasury/types.go

```diff
@@ -0,0 +1,156 @@
+package treasury
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/decred/dcrd/blockchain/stake/v4"
+	"github.com/decred/dcrdata/exchanges/v2"
+	"github.com/decred/dcrdata/v7/db/dbtypes"
+)
+
+// TxParams models the treasury transactions post data structure.
+type TxParams struct {
+	Limit  int64        `json:"limit"`
+	Offset int64        `json:"offset"`
+	TxType stake.TxType `json:"txtype"`
+}
+
+//  Txns models treasury transactions
+type Txns struct {
+	Txns []*TreasuryTx `json:"txns"`
+}
+
+type TreasuryTx struct {
+	TxID        string  `json:"TxID"`
+	Type        int     `json:"Type"`
+	Amount      int64   `json:"Amount"`
+	BlockHash   string  `json:"BlockHash"`
+	BlockHeight int64   `json:"BlockHeight"`
+	BlockTime   TimeDef `json:"BlockTime"`
+}
+
+// Balance models treasury balance.
+type Balance struct {
+	Balance *dbtypes.TreasuryBalance `json:"balance"`
+}
+
+// A page number has the information necessary to create numbered pagination
+// links.
+type PageNumber struct {
+	Active bool   `json:"active"`
+	Link   string `json:"link"`
+	Str    string `json:"str"`
+}
+
+type PageNumbers []PageNumber
+
+type TreasuryInfo struct {
+	Net string
+
+	// Page parameters
+	MaxTxLimit    int64
+	Path          string
+	Limit, Offset int64  // ?n=Limit&start=Offset
+	TxnType       string // ?txntype=TxnType
+
+	// TODO: tadd and tspend can be unconfirmed. tspend for a very long time.
```

Comment:
Do we need to create an issue for this?

Reply:
No we don't to, that comment is from upstream. I left it there for reference purpose.